src/patient-registration/patient-relationship-registration.test.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { render, fireEvent, wait, screen, cleanup } from '@testing-library/react';
+import { PatientRegistration } from './patient-registration.component';
+import { mockUniquePatientIdentifiers } from '../../__mocks__/uniquePatientIdentifiers.mock';
+import { mockCurrentUserLocation } from '../../__mocks__/currentUserLocation.mock';
+import { mockPersons } from '../../__mocks__/persons.mock';
+import { mockRelationshipTypes } from '../../__mocks__/relationshipTypes.mock';
+import {
+  getAllRelationshipTypes,
+  getCurrentUserLocation,
+  getUniquePatientIdentifier,
+  getPerson,
+} from './patient-registration.resource';
+
+let wrapper: any;
+const mockgetAllRelationshipTypes = getAllRelationshipTypes as jest.Mock;
+const mockgetCurrentUserLocation = getCurrentUserLocation as jest.Mock;
+const mockgetUniquePatientIdentifier = getUniquePatientIdentifier as jest.Mock;
+const mockgetPerson = getPerson as jest.Mock;
+
+jest.mock('./patient-registration.resource', () => ({
+  getAllRelationshipTypes: jest.fn(),
+  getCurrentUserLocation: jest.fn(),
+  getUniquePatientIdentifier: jest.fn(),
+  getPerson: jest.fn(),
+}));
+
+const mockPersonResult = mockPersons;
+const mockRelationshipTypesResult = mockRelationshipTypes;
+const mockCurrentUserLocationResult = mockCurrentUserLocation;
+const mockUniquePatientIdentifierResult = mockUniquePatientIdentifiers;
+
+describe('relationship section', () => {
+  afterEach(cleanup);
+  beforeEach(mockgetAllRelationshipTypes.mockReset);
+  beforeEach(mockgetCurrentUserLocation.mockReset);
+  beforeEach(mockgetUniquePatientIdentifier.mockReset);
+  beforeEach(mockgetPerson.mockReset);
+
+  function initMocks() {
+    mockgetAllRelationshipTypes.mockReturnValue(Promise.resolve(mockRelationshipTypesResult));
+    mockgetCurrentUserLocation.mockReturnValue(Promise.resolve(mockCurrentUserLocationResult));
+    mockgetUniquePatientIdentifier.mockReturnValue(Promise.resolve(mockUniquePatientIdentifierResult));
+    mockgetPerson.mockReturnValue(Promise.resolve(mockPersonResult));
+  }
+  it('updates to correct relationship name', async () => {
+    initMocks();
+    wrapper = render(<PatientRegistration />);
+    const personNameInput = screen.getByPlaceholderText('Person name') as HTMLInputElement;
+
+    const expectedRelationships = [
+      { uuid: '768-987', name: 'odora hill', type: '8d919b58-c2cc-11de-8d13-0010c6dffd0f-A' },
+    ];
+
+    await wait(() => {
+      fireEvent.change(personNameInput, { target: { value: expectedRelationships[0].name } });
+    });
+    expect(personNameInput.value).toEqual(expectedRelationships[0].name);
```

Comment:
Neither of these tests actually test anything. Here are some things you might want to test:

- when a few letters are entered, a patient with a corresponding name appears in the suggestion list
- when a few more letters are added, the suggestion list updates again
- when a suggestion is clicked, something changes visually to indicate that the patient is selected (I haven't seen the functionality, so I don't know what it is)
- after adding a patient relationship, another can be added
- when submitting a form with relationships, the expected relationship data is included in the request body

You don't have to implement all of these, but they would be examples of tests that would be valuable, and you should implement at least one or two. Tests that test nothing are worse than no tests at all.